libtesting-ui/src/main/java/com/mapbox/navigation/testing/ui/BaseTest.kt

```diff
@@ -0,0 +1,51 @@
+package com.mapbox.navigation.testing.ui
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.ActivityTestRule
+import androidx.test.uiautomator.UiDevice
+import com.mapbox.navigation.testing.ui.http.MockWebServerRule
+import com.schibsted.spain.barista.rule.cleardata.ClearDatabaseRule
+import com.schibsted.spain.barista.rule.cleardata.ClearFilesRule
+import com.schibsted.spain.barista.rule.cleardata.ClearPreferencesRule
+import org.junit.Rule
+
+open class BaseTest<A : AppCompatActivity>(activityClass: Class<A>) {
+
+    @get:Rule
+    val activityRule = ActivityTestRule(activityClass)
+
+    @get:Rule
+    val mockLocationUpdatesRule = MockLocationUpdatesRule()
+
+    @get:Rule
+    val mockWebServerRule = MockWebServerRule()
+
+    // Clear all app's SharedPreferences
+    @get:Rule
+    val clearPreferencesRule = ClearPreferencesRule()
+
+    // Delete all tables from all the app's SQLite Databases
```

Comment:
Want to change the comment to be, why we are doing this? The "ClearDatabaseRule" explains that it deletes the app's database tables.

But I'm not sure what would happen if we removed the rule. Would it break the tests? Is there a telemetry database or something?

Reply:
This helps to run the tests in isolation. Maps SDK resources are stored in an SQLite database which gets cleared before each test thanks to this rule. I added an inline comment.